src/hooks/useServerSideCurrentUser.ts

```diff
@@ -0,0 +1,27 @@
+import "server-only";
+
+import { ConfigService } from "@nestjs/config";
+import { JwtScope } from "app/api/authentication/enums/jwt-scope.enum";
+import { JwtSessionCookie } from "app/api/authentication/interfaces/jwt-session-cookie.interface";
+import { JwtService } from "app/api/authentication/jwt.service";
+import { User } from "app/api/user/models/user.model";
+import { UserService } from "app/api/user/user.service";
+import { useService } from "app/hooks/useService";
+import { cookies } from "next/headers";
+
+export async function useServerSideCurrentUser(): Promise<User> {
```

Comment:
I'd prefer it if we don't use the `use` prefix here since it resembles react hooks too much and makes it harder to find the boundaries between server and client components (that can already be confusing)